test/functional/wallet_dump.py

```diff
@@ -209,6 +209,15 @@ def run_test(self):
         with self.nodes[0].assert_debug_log(['Flushing wallet.dat'], timeout=20):
             self.nodes[0].getnewaddress()
 
+        # Make sure that dumpwallet doesn't have a lock order issue when there is an unconfirmed tx and it is reloaded
+        # See https://github.com/bitcoin/bitcoin/issues/22489
+        self.nodes[0].createwallet("w3")
+        w3 = self.nodes[0].get_wallet_rpc("w3")
+        w3.generatetoaddress(101, w3.getnewaddress())
```

Comment:
Wouldn't this be faster?

```py
w3.importprivkey(privkey=self.nodes[0].get_deterministic_priv_key().key, label='coinbase_import')

Reply:
For some reason I thought this was a clean chain. Changed as suggested.